src/test/mina_automation/daemon.ml

```diff
@@ -0,0 +1,186 @@
+(**
+Module to run daemon process.
+*)
+open Core
+
+open Async
+
+module Paths = struct
+  let dune_name = "src/app/cli/src/mina.exe"
+
+  let official_name = "mina"
+end
+
+module Executor = Executor.Make (Paths)
+
+type t = Executor.t
+
+let logger = Logger.create ()
+
+(** 
+  Module [Client] provides functions to interact with a Mina daemon.
+*)
+module Client = struct
+  type t = { port : int; executor : Executor.t }
+
+  let create ?(port = 3085) ?(executor = Executor.AutoDetect) () =
+    { port; executor }
+
+  (** [stop_daemon t] stops the daemon running on the specified port.
+    @param t The daemon instance containing the executor and port information.
+    @return Unit. Executes the command to stop the daemon using the executor.
+  *)
+  let stop_daemon t =
+    Executor.run t.executor
+      ~args:[ "client"; "stop-daemon"; "-daemon-port"; sprintf "%d" t.port ]
```

Comment:
Once we change any CLI arg we basically break all tests. So, I think it would be nice if we could track these together. 

I understand this is hard to do, because we 're using a CLI library that has custom DSL. So I don't expect this to actually happen.